# Cross Validation

Con el fin obtener un mejor performance del modelo se realiza una validación Cruzada, esto con el fin de identificar que modelos pueden llegar a tener un mejor desempeño.

Para esto se define una muestra base inicial, con el fin de que las validaciones puedan reproducibles con resultados iguales.

In [28]:
import warnings
warnings.simplefilter("ignore")

In [29]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.model_selection import cross_val_score #Libreria utilizada para hacer la validación cruzada
from sklearn.linear_model import Lasso

#### Se debe tener en cuenta que los datos utilizados en este script contienen las nuevas columnas obtenidas en la etapa de selection features. Adicionalmente se ha realizado una reducción en el volumen de datos esto debido a que en muchos casos se desconocia la cantidad de cines donde la pelicula se habia estrenado.

In [30]:
X = pd.read_csv(r'C:\Scripts\Proyectos\Predicción_de_ingresos_peliculas\Data\X_opening_p.csv')
y = X['worldwide_gross']
X = X.drop('worldwide_gross',axis=1)

In [31]:
X_train, X_test, y_train, y_test = train_test_split(X,y,random_state=1) #El parametro random_state define cual de los escenarios se va a llamar de forma igual siempre.

In [32]:
scores = cross_val_score(Lasso(),X,y,cv=5,scoring='r2') #En la configuración de la función se define la cantidad de veces que queremos dividir los datos así como la medida que se quiere estimar, en este caso r2.
scores

array([0.59316596, 0.68931527, 0.55383855, 0.18147236, 0.23040894])

In [33]:
scores.mean()

0.44964021745791716

De esto se obtiene un score promedio de 0.44, de esto se procede a comparar con el error generado por el metodo tradicional.

In [35]:
Lasso().fit(X_train,y_train).score(X_test,y_test)

0.7926190294963362

De esto se establece que el Score robusto para el modelo sera de 0.4496 y el score convencional sera de 0.7926